### PR TITLE
FIX: play audio sound on message in non group DMs

### PR DESCRIPTION
**Pull request by jjaffeux**
Prior to this change, only mentions would get a notification and a sound. This change will not create a notification for this case, but will play a sound. This is still respecting notification settings, not playing the sound when you are viewing the channel or not following it.

### CODE REVIEW

File: plugins/chat/assets/javascripts/discourse/services/chat-audio-manager.js [hunk 1]
```diff
@@ -37,11 +37,13 @@ export default class ChatAudioManager extends Service {
   }
 
   async play(soundName) {
+    if (isTesting()) {
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/services/chat-audio-manager.js:40]: we can't test audio anyways, so rather exit early

File: plugins/chat/assets/javascripts/discourse/services/chat-subscriptions-manager.js [hunk 2]
```diff
@@ -205,6 +206,8 @@ export default class ChatSubscriptionsManager extends Service {
             channel.tracking.unreadCount++;
           }
 
+          this.chatChannelNotificationSound.play(channel);
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/services/chat-subscriptions-manager.js:209]: this specific line is not tested as it's hard to test an actual sound is played or even this function is called in this case, however the underlying service is fully tested

File: plugins/chat/assets/javascripts/discourse/services/chat-channel-notification-sound.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    const membership = channel.currentUserMembership;
+    if (
+      !membership.following ||
+      membership.desktopNotificationLevel !== "always" ||
+      membership.muted ||
+      this.chat.activeChannel === channel
+    ) {
+      return false;
+    }
```
Comment (ZogStriP) [inline plugins/chat/assets/javascripts/discourse/services/chat-channel-notification-sound.js:38]: I wonder if this is easier to read/understand? 🤔 

```suggestion
    if (!membership.following) {
      return false;
    }
    
    if (membership.desktopNotificationLevel !== "always") {
      return false;
    }
    
    if (membership.muted) {
      return false;
    }
    
    if (this.chat.activeChannel === channel) {
      return false;
    }
```

File: plugins/chat/test/javascripts/unit/services/chat-channel-notification-sound-test.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    needs.user();
+
+    test("in do not disturb", async function (assert) {
+      updateCurrentUser({ do_not_disturb_until: moment().add(1, "hour") });
+      const channel = buildDirectMessageChannel(getOwner(this));
+
+      assert.deepEqual(await this.subject.play(channel), false);
+    });
+
+    test("not chat sound", async function (assert) {
```
Comment (ZogStriP) [inline plugins/chat/test/javascripts/unit/services/chat-channel-notification-sound-test.js:58]: Typo?

```suggestion
    test("no chat sound", async function (assert) {
```

File: plugins/chat/test/javascripts/unit/services/chat-channel-notification-sound-test.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    });
+
+    test("mobile", async function (assert) {
+      const channel = buildDirectMessageChannel(getOwner(this));
+      this.site.mobileView = true;
+
+      assert.deepEqual(await this.subject.play(channel), false);
+    });
+
+    test("default", async function (assert) {
```
Comment (ZogStriP) [inline plugins/chat/test/javascripts/unit/services/chat-channel-notification-sound-test.js:72]:
```suggestion
    test("plays sound", async function (assert) {
```

**Review by ZogStriP — APPROVED** [review]
🔊

**Pull request merged**